### PR TITLE
Fix and modernize documentation and repo links

### PR DESCRIPTION
**Pull request by TheJCAB**
- Replace "master" with "HEAD", which always references the tip of the default branch.
- Rename docs.microsoft.com to learn.microsoft.com (apparently they couldn't leave good enough alone).
- Remove "en-us" from docs/lean links, so the user gets their default locale (suggestion from Don).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,13 +3,13 @@ This is a collection of extensions and sample scripts for extending WinDbg. We'l
 # Getting Started
 To load JavaScript extensions:
 1. Download the script file locally.
-2. Ensure you have a recent version of WinDbg - [WinDbg Preview](https://aka.ms/WinDbgPreview) from the Microsoft Store will always be up to date. Otherwise you can use one of the other methods listed [here](https://docs.microsoft.com/en-us/windows-hardware/drivers/debugger/) to install it.
+2. Ensure you have a recent version of WinDbg - [WinDbg Preview](https://aka.ms/WinDbgPreview) from the Microsoft Store will always be up to date. Otherwise you can use one of the other methods listed [here](https://docs.microsoft.com/windows-hardware/drivers/debugger/) to install it.
```
Comment (lzybkr) [inline README.md:6]: We're out of preview, this should get updated.

File: README.md [hunk 2]
```diff
@@ -3,13 +3,13 @@ This is a collection of extensions and sample scripts for extending WinDbg. We'l
...
     * If JavaScript isn't on the list, run `.load jsprovider`
 5. Run `.scriptload <path to script>` or `.scriptrun <path to script>`. The README for each script has more detailed usage information.
 
-We have more information on our JavaScript support at https://docs.microsoft.com/en-us/windows-hardware/drivers/debugger/javascript-debugger-scripting. If you want to learn more about a script you can open in it WinDbg Preview by hitting the "Scripting" ribbon and clicking "Open Script...". It has intellisense support for JavaScript and NatVis and you can load scripts by hitting "Execute" in the scripting ribbon.
+We have more information on our JavaScript support at https://docs.microsoft.com/windows-hardware/drivers/debugger/javascript-debugger-scripting. If you want to learn more about a script you can open in it WinDbg Preview by hitting the "Scripting" ribbon and clicking "Open Script...". It has intellisense support for JavaScript and NatVis and you can load scripts by hitting "Execute" in the scripting ribbon.
```
Comment (lzybkr) [inline README.md:12]: Another mention of preview that should get updated.

**Review by lzybkr — COMMENTED** [review]
I commented on a couple references to WinDbg Preview, but there are more. Can you do another pass to clean those up?

File: ImageInfo/README.MD [hunk 3]
```diff
@@ -9,8 +9,8 @@ Run `.scriptload ImageInfo.js` to load the script. Any module accessed via `dx`,
 This extension extends the debugger's concept of a module to include:
 * <b>Headers</b> - A list of available headers from the PE image for the module.
     * <b>ImageHeader</b> - The DOS header for the module -- an _IMAGE_DOS_HEADER.
-    * <b>FileHeader</b> - The file header for the module -- an [_IMAGE_FILE_HEADER](https://docs.microsoft.com/en-us/windows/desktop/api/winnt/ns-winnt-_image_file_header).
-    * <b>OptionalHeader</b> - The optional header for the module -- an [_IMAGE_OPTIONAL_HEADER(64)](https://docs.microsoft.com/en-us/windows/desktop/api/winnt/ns-winnt-_image_optional_header64).
+    * <b>FileHeader</b> - The file header for the module -- an [_IMAGE_FILE_HEADER](https://docs.microsoft.com/windows/desktop/api/winnt/ns-winnt-_image_file_header).
+    * <b>OptionalHeader</b> - The optional header for the module -- an [_IMAGE_OPTIONAL_HEADER(64)](https://docs.microsoft.com/windows/desktop/api/winnt/ns-winnt-_image_optional_header64).
```
Comment (KenSykes) [inline ImageInfo/README.MD:13]: these links need the en-us, the new ones are broken

File: TTD/docs/TTDLiveRecorder.h/interface-ILiveRecorder.md [hunk 4]
```diff
@@ -30,9 +30,9 @@ may be used to ensure lifetime is handled correctly.
 
 These methods control the access and lifetime of the object that provides the interface.
 
-- [`AddRef`](https://learn.microsoft.com/en-us/windows/win32/api/unknwn/nf-unknwn-iunknown-addref) (IUnknown)
-- [`Release`](https://learn.microsoft.com/en-us/windows/win32/api/unknwn/nf-unknwn-iunknown-release) (IUnknown)
-- [`QueryInterface`](https://learn.microsoft.com/en-us/windows/win32/api/unknwn/nf-unknwn-iunknown-queryinterface(refiid_void)) (IUnknown)
+- [`AddRef`](https://learn.microsoft.com/windows/win32/api/unknwn/nf-unknwn-iunknown-addref) (IUnknown)
+- [`Release`](https://learn.microsoft.com/windows/win32/api/unknwn/nf-unknwn-iunknown-release) (IUnknown)
+- [`QueryInterface`](https://learn.microsoft.com/windows/win32/api/unknwn/nf-unknwn-iunknown-queryinterface(refiid_void)) (IUnknown)
```
Comment (KenSykes) [inline TTD/docs/TTDLiveRecorder.h/interface-ILiveRecorder.md:36]: this one needs en-us

File: TTDQueries/README.md [hunk 5]
```diff
@@ -48,6 +48,6 @@ Check out [TTD - Queries] to learn more.
 * [Defrag Tools - Time Travel Debugging - Advanced](https://channel9.msdn.com/Shows/Defrag-Tools/Defrag-Tools-186-Time-Travel-Debugging-Advanced)
```
Comment (KenSykes) [inline TTDQueries/README.md:48]: these links need to be updated:


[https://learn.microsoft.com/en-us/shows/defrag-tools/185-time-travel-debugging-introduction](https://learn.microsoft.com/en-us/shows/defrag-tools/185-time-travel-debugging-introduction)

[https://learn.microsoft.com/en-us/shows/defrag-tools/186-time-travel-debugging-advanced](https://learn.microsoft.com/en-us/shows/defrag-tools/186-time-travel-debugging-advanced)

File: TTD/docs/TTDLiveRecorder.h/interface-ILiveRecorder.md (hunk 4 above)
Comment (TheJCAB) [reply TTD/docs/TTDLiveRecorder.h/interface-ILiveRecorder.md:36]: No it doesn't? Just tried it, works fine.

Comment (TheJCAB) [reply TTD/docs/TTDLiveRecorder.h/interface-ILiveRecorder.md:36]: Note I removed "en-us" following @DOMARS instructions.

Comment (TheJCAB) [reply TTD/docs/TTDLiveRecorder.h/interface-ILiveRecorder.md:36]: Aha! The parenthesis are confusing CodeFlow. All others (VSCode and GitHub) seem to be able to handle it fine.

I'll escape it anyway.

File: ImageInfo/README.MD (hunk 3 above)
Comment (TheJCAB) [reply ImageInfo/README.MD:13]: Actually, they truly are broken. "en-us" is not related to that.

**Comment by DOMARS** [comment]
Hi @TheJCAB,
Jason asked that I give this a look over. Looks good to me! 
(I too was OK with docs.microsoft, and even MSDN, but thanks for updating to the latest "learn" URLs.)

**Comment by DOMARS** [comment]
@lzybkr - Looks like this PR is awaiting your review. The link updates look good to me. 
Don

**Review by lzybkr — APPROVED** [review]

**Pull request merged**